src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/Record2SolutionMappingTranslatorImpl.java

```diff
@@ -0,0 +1,128 @@
+            } else if (value instanceof LPGEdgeValue) {
+                final Node edgeMapping = conf.mapEdgeLabel(((LPGEdgeValue) value).getEdge().getLabel());
+                if (builders.isEmpty())
+                    baseBuilder.add(mappingVar, edgeMapping);
+                else {
+                    addToAllBuilders(builders, mappingVar, edgeMapping);
+                }
+            } else if (value instanceof MapValue) {
+                final Map<String, Object> mapValue = ((MapValue) value).getMap();
+                final Node triple = NodeFactory.createTripleNode(
+                        conf.mapNode((LPGNode) mapValue.get("source")),
+                        conf.mapEdgeLabel(((LPGEdge) mapValue.get("edge")).getLabel()),
+                        conf.mapNode((LPGNode) mapValue.get("target"))
+                );
+                if (builders.isEmpty())
+                    baseBuilder.add(mappingVar, triple);
+                else {
+                    addToAllBuilders(builders, mappingVar, triple);
+                }
+            } else if (value instanceof ListValue) {
+                final List<Object> values = ((ListValue) value).getList();
+                final boolean isPropertyList = CypherUtils.isPropertyColumn(query, var);
+                if (builders.isEmpty()) {
+                    for (final Object val : values) {
+                        Node node;
+                        if (isPropertyList) {
+                            node = conf.mapProperty(val.toString());
+                        } else {
+                            node = NodeFactory.createLiteral(val.toString());
+                        }
+                        final BindingBuilder builder = Binding.builder();
+                        builder.addAll(baseBuilder.build());
+                        builder.add(mappingVar, node);
+                        builders.add(builder);
+                    }
+                } else {
+                    for (final Object val : values) {
+                        Node node;
+                        if (isPropertyList) {
+                            node = conf.mapProperty(val.toString());
+                        } else {
+                            node = NodeFactory.createLiteral(val.toString());
+                        }
+                        addToAllBuilders(builders, mappingVar, node);
+                    }
+                }
+            } else if (value instanceof LiteralValue) {
+                final String literal = ((LiteralValue) value).getValue().toString();
+                Node node;
+                if (literal.equals("label"))
+                    node = conf.getLabel();
+                else
+                    node = NodeFactory.createLiteral(literal);
+                if (builders.isEmpty()) {
+                    baseBuilder.add(mappingVar, node);
+                } else {
+                    addToAllBuilders(builders, mappingVar, node);
+                }
+            }
```

Comment:
There is no else-branch here. That's not good. Never assume that, only because the else-if branches cover all the cases that currently exist, there won't be any problem. Instead, always assume that further cases may be added in the future and, therefore, we need to put safeguards into place that give us explicit errors if we forget to cover these future cases. There are two ways to do this: First, you convert the last else-if into an else branch; that way, we would see an exception when cast `value` to a `LiteralValue`. Second, add an else branch in which you throw an IllegalArgumentException.

As a general rule: always make sure you have an else branch after else-if branches.